core/src/main/scala/cats/data/WriterT.scala

```diff
@@ -0,0 +1,73 @@
+package cats
+package data
+
+final case class WriterT[T[_], L, V] (run: T[(L, V)]) {
+  def written (implicit functorT: Functor[T]): T[L] =
+    functorT.map (run)(_._1)
+
+  def value (implicit functorT: Functor[T]): T[V] =
+    functorT.map (run)(_._2)
+
+  def map[Z](fn: V => Z)(implicit functorT: Functor[T]): WriterT[T, L, Z] =
+    WriterT {
+      functorT.map (run) { z =>
+        (z._1, fn (z._2))
+      }
+    }
+
+  def flatMap[U](f: V => WriterT[T, L, U])(implicit monadT: Monad[T], semigroupL: Semigroup[L]): WriterT[T, L, U] =
+    WriterT {
+      monadT.flatMap (run) { lv =>
+        monadT.map (f (lv._2).run) { lv2 =>
+          (semigroupL.combine (lv._1, lv2._1), lv2._2)
+        }
+      }
+    }
+
+  def mapValue[M, U] (f: ((L, V)) => (M, U))(implicit functorT: Functor[T]): WriterT[T, M, U] =
+    WriterT { functorT.map (run)(f) }
+
+  def mapWritten[M] (f: L => M)(implicit functorT: Functor[T]): WriterT[T, M, V] =
+    mapValue (wa => (f (wa._1), wa._2))
+
+  def swap (implicit functorT: Functor[T]): WriterT[T, V, L] =
+    mapValue (wa => (wa._2, wa._1))
+
+  def reset (implicit monoidL: Monoid[L], functorT: Functor[T]): WriterT[T, L, V] =
+    mapWritten (_ => monoidL.empty)
+}
+object WriterT extends WriterTInstances with WriterTFunctions
+
+sealed abstract class WriterTInstances {
+  implicit def writerTMonad[T[_], L] (implicit monadT: Monad[T], monoidL: Monoid[L]) = {
+    new Monad[({type WT[$] = WriterT[T, L, $]})#WT] {
```

Comment:
Would you mind converting these to use [kind-projector](https://github.com/non/kind-projector) to be consistent with other Cats code?